overlord/snapstate/handlers.go

```diff
@@ -503,24 +503,45 @@ func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq map[stri
 		}
 	}
 
+	// If transactional, use a single lane for all tasks, so when
+	// one fails the changes for all affected snaps will be
+	// undone. Otherwise, have different lanes per snap so
+	// failures only affect the culprit snap.
+	var transactionLane int
+	if flags.Transactional {
+		transactionLane = st.NewLane()
```

Comment:
Shouldn't this lane be shared between the prerequisites and the current task `t`? Or is it intentional that the prerequisites are only in a transaction among themselves but not the "main" task?

Reply:
Good catch, yes, it needs to use the lane that already exists for the task. I've changed that now.

Reply:
Thanks. We should add a test to cover this, like doing an install that triggers a prerequisite and then check its lane

Reply:
Well, actually your comment made me think that something was missing, that was testing that pre-requisites really joined the existing lane when running, for instance, `InstallMany`. And that helped me identify a bug, so thanks for this :-)

Please take a look at my new commits.